api-reference/10 UI Widgets/dxDiagram/1 Configuration/customShapeToolboxTemplate.md

```diff
@@ -4,16 +4,25 @@ type: template
 ---
 ---
 ##### shortDescription
-<!-- Description goes here -->
+Specifies a custom template for shapes in the toolbox.
 
 ##### param(container): dxSVGElement
-<!-- Description goes here -->
+#include common-ref-elementparam with { element: "shape" }
 
 ##### param(data): Object
-<!-- Description goes here -->
+Information about the currently processed shape.
 
 ##### field(data.item): dxDiagramShape
-<!-- Description goes here -->
+Information about the currently processed shape.
 
 ---
-<!-- Description goes here -->
\ No newline at end of file
+The template content must be presented as SVG elements.
+
+Use the [toolboxTemplate](/api-reference/10%20UI%20Widgets/dxDiagram/1%20Configuration/customShapes/toolboxTemplate.md '/Documentation/ApiReference/UI_Widgets/dxDiagram/Configuration/customShapes/#toolboxTemplate') option to define a template of an individual shape in the toolbox.
```

Comment:
Can you clarify what you mean by "a template of an individual shape"?

Reply:
Diagram has two options for a shape template in the toolbox: one specifies a template for all shapes (diagram.customShapeToolboxTemplate). Another specifies a template for a particular shape (customShape.toolboxTemplate), so users can create a template for every shape.

Reply:
```suggestion
Use the [toolboxTemplate](/api-reference/10%20UI%20Widgets/dxDiagram/1%20Configuration/customShapes/toolboxTemplate.md '/Documentation/ApiReference/UI_Widgets/dxDiagram/Configuration/customShapes/#toolboxTemplate') option to define a template for each shape in the toolbox.
```